network-fs.go

```diff
@@ -0,0 +1,239 @@
+	httpClient *http.Client
+}
+
+const (
+	connected       = "200 Connected to Go RPC"
+	dialTimeoutSecs = 30 // 30 seconds.
+)
+
+// splits network path into its components Address and Path.
+func splitNetPath(networkPath string) (netAddr, netPath string) {
+	index := strings.LastIndex(networkPath, ":")
+	netAddr = networkPath[:index]
+	netPath = networkPath[index+1:]
+	return netAddr, netPath
+}
+
+// Initialize new network file system.
+func newNetworkFS(networkPath string) (StorageAPI, error) {
+	// Input validation.
+	if networkPath == "" && strings.LastIndex(networkPath, ":") != -1 {
+		return nil, errInvalidArgument
+	}
+
+	// TODO validate netAddr and netPath.
+	netAddr, netPath := splitNetPath(networkPath)
+
+	// Dial minio rpc storage http path.
+	rpcClient, err := rpc.DialHTTPPath("tcp", netAddr, "/minio/rpc/storage")
+	if err != nil {
+		return nil, err
+	}
+
+	// Initialize http client.
+	httpClient := &http.Client{
+		// Setting a sensible time out of 6minutes to wait for
+		// response headers. Request is pro-actively cancelled
+		// after 6minutes if no response was received from server.
+		Timeout:   6 * time.Minute,
+		Transport: http.DefaultTransport,
+	}
+
+	// Initialize network storage.
+	ndisk := &networkFS{
+		netAddr:    netAddr,
+		netPath:    netPath,
+		rpcClient:  rpcClient,
+		httpClient: httpClient,
+	}
+
+	// Returns successfully here.
+	return ndisk, nil
+}
+
+// MakeVol - make a volume.
+func (n networkFS) MakeVol(volume string) error {
+	reply := GenericReply{}
+	if err := n.rpcClient.Call("Storage.MakeVolHandler", volume, &reply); err != nil {
+		if err.Error() == errVolumeExists.Error() {
+			return errVolumeExists
+		}
+		return err
+	}
+	return nil
+}
+
+// ListVols - List all volumes.
+func (n networkFS) ListVols() (vols []VolInfo, err error) {
+	ListVols := ListVolsReply{}
+	err = n.rpcClient.Call("Storage.ListVolsHandler", "", &ListVols)
+	if err != nil {
+		return nil, err
+	}
+	return ListVols.Vols, nil
+}
+
+// StatVol - get current Stat volume info.
+func (n networkFS) StatVol(volume string) (volInfo VolInfo, err error) {
+	if err = n.rpcClient.Call("Storage.StatVolHandler", volume, &volInfo); err != nil {
+		if err.Error() == errVolumeNotFound.Error() {
+			return VolInfo{}, errVolumeNotFound
+		}
+		return VolInfo{}, err
+	}
+	return volInfo, nil
+}
+
+// DeleteVol - Delete a volume.
+func (n networkFS) DeleteVol(volume string) error {
+	reply := GenericReply{}
+	if err := n.rpcClient.Call("Storage.DeleteVolHandler", volume, &reply); err != nil {
+		if err.Error() == errVolumeNotFound.Error() {
+			return errVolumeNotFound
+		}
+		return err
+	}
+	return nil
+}
+
+// File operations.
+
+// CreateFile - create file.
+func (n networkFS) CreateFile(volume, path string) (writeCloser io.WriteCloser, err error) {
+	writeURL := new(url.URL)
+	writeURL.Scheme = "http" // TODO fix this.
+	writeURL.Host = n.netAddr
+	writeURL.Path = fmt.Sprintf("/minio/rpc/storage/upload/%s", urlpath.Join(volume, path))
```

Comment:
Please define a constant for the path `/minio/rpc/storage/upload` and use `urlpath.Join()` than `fmt.Sprintf()`